src/svm/interface/svm.c

```diff
@@ -1441,6 +1441,52 @@ PetscErrorCode SVMView(SVM svm,PetscViewer v)
   PetscFunctionReturn(0);
 }
 
+/*@
+  SVMViewTrainingPredictions - Views predictions on training samples using trained model.
+
+  Input Parameters:
++ svm - SVM context
+- v - visualization context
+
+  Level: beginner
+
+.seealso SVMLoadTrainingDataset, SVMPredict, PetscViewer
+@*/
+PetscErrorCode SVMViewTrainingPredictions(SVM svm,PetscViewer v)
+{
+
+  PetscFunctionBeginI;
+  PetscValidHeaderSpecific(svm,SVM_CLASSID,1);
+  PetscValidHeaderSpecific(v,PETSC_VIEWER_CLASSID,2);
+  if (svm->ops->viewtrainingpredictions) {
+    TRY( svm->ops->viewtrainingpredictions(svm,v) );
+  }
+  PetscFunctionReturnI(0);
```

Comment:
```suggestion
  PetscFunctionReturn(0);
```

Reply:
As above.

Reply:
> You should use PETSc's -log_trace (because of granularity) or proper debugging tools if your program is getting stuck. PERMON's trace is a bit overused in the main library as well - it should trace only really high-level functions like QPT* and QPSSolve.

If it is a software issue, I agree. If it is a hardware or a SaaS issue, it is good to have traced I/O operations. Moreover, loading dataset(s) is traced, thus this would be traced as well for consistency.